handler.js

```diff
@@ -0,0 +1,185 @@
+'use strict';
+
+var AWS = require('aws-sdk');
+var attr = require('dynamodb-data-types').AttributeValue;
+var craigslistApartments = require('./craigslist-apartments');
+var filters = require('./filters');
+
+
+var ddb = new AWS.DynamoDB({apiVersion: '2012-08-10'});
+
+module.exports.hello = (event, context) => {
+  const filter = new filters(event.filters.postalCode, event.filters.kmFromPostalCode, event.filters.minPrice, event.filters.maxPrice, event.filters.minSqft);
+  var apartmentsPromise = craigslistApartments.getListOfRentals(filter);
+  return apartmentsPromise.then((result) => {
+    return result;
+  })
+  .then((apartmentListings) => {
+    return storeInDB(ddb, apartmentListings);
+  })
+  .then((todaysListings) => {
+    var yesterdayListingsPromise = getYesterdaysListings(ddb);
+    return yesterdayListingsPromise.then((result) => {
+      return result;
+    })
+    .then((yesterdayListings) => {
+      var apartmentListings = {
+        todaysListings: todaysListings,
+        yesterdayListings: yesterdayListings
+      }
+      return apartmentListings;
+    })
+    
+  })
+  .then((apartmentListings) => {
+    var apartments = findNewApartments(apartmentListings);
+    var removed = findRemovedApartments(apartmentListings);
+    var message = createdMessage(apartments, removed);
+    return sendEmail(message);
+  })
+  .then(() => {
+    return {
+      statusCode: 200,
+      body: JSON.stringify({ message: "Lambda executed with success" })
+    }
+  })
+}
+
+function findNewApartments(apartmentListings) {
```

Comment:
findNewApartments and findRemovedApartments are essentially the same method: given two lists (listA and listB), you want to know what's in listA that's not in listB.

For new listings, listA is todaysListings and listB is yesterdaysListings; For removed apartments, listA is yesterdaysListings and listB is todaysListings.

Reply:
Done.